Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+PathComponent.swift

```diff
@@ -37,88 +37,444 @@ extension PathHierarchy {
         /// The full original path component
         let full: String
         /// The parsed entity name
-        let name: String
+        let name: Substring
         /// The parsed entity kind, if any.
-        var kind: String?
+        var kind: Substring?
         /// The parsed entity hash, if any.
-        var hash: String?
+        var hash: Substring?
     }
     
+    enum PathParser {
```

Comment:
Here I mainly did it to avoid the extra indentation. By using `extension OuterType.InnerType { ... }` the functions defined on `InnerType` are only indented once instead of twice. This mostly doesn't matter unless the API is added to a deeply nested type. 

However, since I was moving API from `PathHierarchy` into `PathHierarchy.PathParser` while also modifying it, I wanted to preserve the unchanged lines in the diff which wouldn't have worked if the whole code block was intended more than it was before.